tests/utils.cpp

```diff
@@ -68,3 +68,16 @@ TEST_CASE("String parsing") {
         CHECK_THROWS_AS(chemfiles::parse<size_t>("9223372036854775808"), chemfiles::Error);
     }
 }
+
+TEST_CASE("scan") {
+    int i = 0;
+    double d = 0;
+    char s[32] = {0};
+
+    chemfiles::scan("3, 4.2 baz", "%i, %lf %31s", &i, &d, s);
+    CHECK(i == 3);
+    CHECK(d == 4.2);
+    CHECK(std::string(s) == "baz");
+
+    CHECK_THROWS_AS(chemfiles::scan("3, 4.2", "%i, %lf %31s", &i, &d, s), chemfiles::Error);
```

Comment:
Probably a new PR, but should the types be checked? What about `nullptr`?

Reply:
what do you mean with "the types be checked"? The types of the variables passed to be filled? 

They are checked by GCC and Clang thanks to `__attribute__((format(scanf, 2, 3)))`:

https://github.com/chemfiles/chemfiles/blob/4ad63a5d3e646fde45dc0899675bee5ec0c0355f/include/chemfiles/utils.hpp#L102-L105

There is however no runtime error, only a compile time warning. w.r.t. nullptr, I'll see what I can do, but I think it is also checked by GCC (at least in the obvious `scan("%f", nullptr)` case)

Reply:
If it's checked by GCC then disregard the comment.

Reply:
so the types are checked by GCC, and passing a null pointer segfault the code, so I think this will be fine